backend/x11/input_device.c

```diff
@@ -20,6 +20,7 @@ static void send_key_event(struct wlr_x11_backend *x11, uint32_t key,
 		enum wlr_key_state st, xcb_timestamp_t time) {
 	struct wlr_event_keyboard_key ev = {
 		.time_msec = time,
+		.time_frac_nsec = (time % 1000) * 1000000,
```

Comment:
Hmm. This doesn't seem right.

Reply:
`(time % 1000)` has a value within 10^3, so we multiple by 10^6 to get nanosecond (10^9) precision.

Reply:
Isn't `time_frac_nsec` between 0 and 999999?

Reply:
No, between 0 and 999999999 (10^10 - 1).

Reply:
Wait, so some information is duplicated between `time_msec` and `time_frac_nsec`? This is pretty confusing.